packages/react-components/react-combobox/src/utils/useTriggerListboxSlots.ts

```diff
@@ -91,9 +94,17 @@ export function useTriggerListboxSlots(
     }, listbox?.onMouseOver),
   );
 
+  const documentOnMouseUp = useEventCallback((ev: MouseEvent) => {
+    if (!listboxRef.current?.contains(ev.target as HTMLElement)) {
+      setOpen(ev as unknown as React.MouseEvent<HTMLElement>, false);
+    }
+    document.removeEventListener('mouseup', documentOnMouseUp);
```

Comment:
I think we should either use `targetDocument` from `useFluent` / the FluentProvider, or `listboxRef.ownerDocument`

Reply:
You're right, I'm getting it from `useFluent` now 😄